src/Data/Vector/Persistent.hs

```diff
@@ -556,6 +438,35 @@ partition p = spToPair . foldl' go (SP empty empty)
 fromList :: [a] -> Vector a
 fromList = F.foldl' snoc empty
 
+-- | O(n) Take @n@ elements starting from the start of the 'Vector'
+take :: Int -> Vector a -> Vector a
+take n v = fromList (L.take n (F.toList v))
+
```

Comment:
I think we must be able to do at least somewhat better than this for `take`. Shouldn't we be able to share at least one subtree?

Reply:
Yup - the easy way is to just set the size to be lower (but also update the folds to stop early). I'll bring that back once I convince myself that I have a correct implementation of the slicing logic.

Reply:
That's easy, but it doesn't release memory. I think it's valuable to write "real" (non-slice) operations as efficiently as possible. `take`, unlike `drop`, smells like a good candidate. 

My concept of how the slicing stuff should go.

1. Have an `Unsliced` module with the base implementation. Gradually add bounded operations.
2. Once there are enough bounded operations to make it useful, add a `Sliced` module defining `data Vector a = Vector {fstidx :: !Int, slicelen :: !Int, vec :: {-#UNPACK#-} !(Unsliced.Vector a)}`. This can use the bounded operations, and add its own wrappers around indexing operations.